src/main/java/de/tum/in/www1/artemis/service/LectureUnitProcessingService.java

```diff
@@ -0,0 +1,228 @@
+     */
+    public List<LectureUnitDTO> splitUnits(LectureUnitInformationDTO lectureUnitInformationDTO, MultipartFile file) throws IOException {
+
+        try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream(); PDDocument document = PDDocument.load(file.getBytes())) {
+            List<LectureUnitDTO> units = new ArrayList<>();
+            Splitter pdfSplitter = new Splitter();
+
+            for (LectureUnitSplitDTO lectureUnit : lectureUnitInformationDTO.units()) {
+                AttachmentUnit attachmentUnit = new AttachmentUnit();
+                Attachment attachment = new Attachment();
+                PDDocumentInformation pdDocumentInformation = new PDDocumentInformation();
+
+                pdfSplitter.setStartPage(lectureUnit.startPage());
+                pdfSplitter.setEndPage(lectureUnit.endPage());
+                pdfSplitter.setSplitAtPage(lectureUnit.endPage());
+
+                List<PDDocument> documentUnits = pdfSplitter.split(document);
+                pdDocumentInformation.setTitle(lectureUnit.unitName());
+                if (lectureUnitInformationDTO.removeBreakSlides() != null && lectureUnitInformationDTO.removeBreakSlides()) {
+                    removeBreakSlides(documentUnits.get(0));
+                }
+                documentUnits.get(0).setDocumentInformation(pdDocumentInformation);
+                documentUnits.get(0).save(outputStream);
+
+                // setup attachmentUnit and attachment
+                attachmentUnit.setDescription("");
+                attachment.setName(lectureUnit.unitName());
+                attachment.setAttachmentType(AttachmentType.FILE);
+                attachment.setReleaseDate(lectureUnit.releaseDate());
+                attachment.setUploadDate(ZonedDateTime.now());
+
+                MultipartFile multipartFile = convertByteArrayToMultipart(lectureUnit.unitName(), outputStream.toByteArray());
+
+                LectureUnitDTO lectureUnitsDTO = new LectureUnitDTO(attachmentUnit, attachment, multipartFile);
+                units.add(lectureUnitsDTO);
+                documentUnits.get(0).close(); // make sure to close the document
+            }
+            document.close();
+            return units;
+        }
+    }
+
+    /**
+     * Removes the break slides from the given document.
+     * @param document         document to remove break slides from
+     */
+    private void removeBreakSlides(PDDocument document) {
+
+        try {
+            PDFTextStripper pdfTextStripper = new PDFTextStripper();
+            Splitter pdfSplitter = new Splitter();
+            List<PDDocument> pages = pdfSplitter.split(document);
+            Iterator<PDDocument> iterator = pages.listIterator();
+
+            int index = 0;
+            while (iterator.hasNext()) {
+                PDDocument currentPage = iterator.next();
+                String slideText = pdfTextStripper.getText(currentPage);
+                if (slideText.contains("Break")) {
```

Comment:
This currently only works for English. Keep in mind Artemis also supports German.

Reply:
Okay, so by including also Pause would be fine?